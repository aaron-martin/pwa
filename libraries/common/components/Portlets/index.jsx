import React from 'react';
import PropTypes from 'prop-types';

/**
 * The Portlets component activates all the extension components that use portals.
 * @param {Object} props The component props.
 * @param {Array} props.components An array of react components.
 * @return {JSX}
 */
const Portlets = ({ components }) => {
  if (!components || !Object.keys(components).length) {
    return null;
  }

  return Object.keys(components).map((key) => {
    const Component = components[key];

    if (!Component || typeof Component !== 'function') {
      return null;
    }

    return <Component key={key} />;
  });
};

Portlets.propTypes = {
  components: PropTypes.shape(),
};

Portlets.defaultProps = {
  components: null,
};

export default Portlets;
