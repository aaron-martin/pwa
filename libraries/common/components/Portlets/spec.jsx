import React from 'react';
import { shallow, mount } from 'enzyme';
import Portlets from './index';

/**
 * A test component.
 * @return {JSX}
 */
const TestComponent = () => (
  <div>My test component</div>
);

describe('Portlets', () => {
  it('should render without any props', () => {
    const wrapper = shallow(<Portlets />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render if array is not set.', () => {
    const wrapper = shallow(<Portlets />);
    expect(wrapper.children().get(0)).toEqual(null);
  });

  it('should not render if array is empty.', () => {
    const wrapper = shallow(<Portlets components={{}} />);
    expect(wrapper.children().get(0)).toEqual(null);
  });

  it('should not render invalid children', () => {
    const wrapper = shallow(<Portlets components={{ sometest: null }} />);
    expect(wrapper.children().get(0)).toEqual(null);
  });

  it('should return an array with the given test components.', () => {
    const componentObject = {
      myTestComponent: TestComponent,
    };
    const wrapper = mount(<Portlets components={componentObject} />);
    expect(wrapper.children.length).toEqual(1);
    expect(wrapper.children().get(0).type === TestComponent).toEqual(true);
  });

  it('should empty the target dom element if is replacer', () => {

  });
});
