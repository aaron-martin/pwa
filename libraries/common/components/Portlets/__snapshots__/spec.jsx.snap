// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portlets should render without any props 1`] = `""`;
